src/js/components/AppListComponent.jsx

```diff
@@ -80,12 +113,42 @@ var AppListComponent = React.createClass({
     });
   },
 
+  getAppsInGroup: function () {
+    var apps = this.state.apps;
+    var currentGroup = this.props.currentGroup;
+
+    return lazy(apps)
+      .filter((app) => app.id.startsWith(currentGroup))
+      .reduce((memo, app) => {
```

Comment:
It took me a while to understand this reduce-function.

Some suggestions or questions on it..
1.
The function-name is still misleading. It pretends to return apps, but it also returns groups.
Is ```getGroupNodes``` maybe better? Every app is in a group. '/' is the root group.

2. 
I didn't understood this group creation. Why is a group created? Think this is also naming..
Maybe ```createGroupStats```, not sure?
In generally this find(), createGroup() and updateGroup() combination looks like another reduce job to me. But I don't have a good suggestion in the moment.

3.
Why a a newly created group is unshifted instead of pushed? That confused me..
Sorting takes places afterwards anyways.